bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkHandler.java

```diff
@@ -69,7 +69,7 @@
      *
      * @return Link may be invalid, but is never null
      */
-    public @NotNull Link getLink(@NotNull Resource resource) {
+    public Optional<Link<Page>> getLink(@NotNull Resource resource) {
```

Comment:
it could be that the reference is not a Page: shouldn't <Page> be dropped?

Reply:
At the moment all these methods defer to `new LinkImpl(url, target, page)` so they can only target a page.

Reply:
Removed in dd1fd25